### PR TITLE
[WRANGLER-47] DataPrep Filter By directive

### PR DESCRIPTION
**Pull request by elfenheart**
Filter By Directive

JIRA: https://issues.cask.co/browse/WRANGLER-47

Note: Currently there is no implementation for **both** filter by condition and filter by value. Unsure what that actually means.

### CODE REVIEW

File: cdap-ui/app/cdap/components/DataPrep/DataPrepErrorAlert/DataPrepErrorAlert.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+.dataprep-error-alert-container {
+  position: absolute;
+  top: 50px;
+  left: 0;
+  width: 100%;
+  background-color: #c0392b;
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepErrorAlert/DataPrepErrorAlert.scss]: Can probably extract this out into `variables.scss` if we're using it as the error color for Data Prep in multiple places?

File: cdap-ui/app/cdap/components/DataPrep/Directives/Filter/index.js [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+                className="form-control"
+                value={this.state.selectedCondition}
+                onChange={this.handleConditionSelect}
+              >
+                {
+                  filterConditions.map((condition) => {
+                    return (
+                      <option
+                        value={condition.filter}
+                        key={condition.id}
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/Directives/Filter/index.js]: I don't think this `id` property exists yet.

File: cdap-ui/app/cdap/components/DataPrep/Directives/Filter/index.js [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    if (this.state.selectedCondition.substr(0, 6) !== 'CUSTOM') { return null; }
+
+    return (
+      <div>
+        <br />
+        <textarea
+          className="form-control"
+          value={this.state.customFilter}
+          onChange={this.handleCustomFilterChange}
+          ref={ref => this.customFilterRef = ref}
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/Directives/Filter/index.js:174]: We're not using this ref right now. Did you want to autofocus on it?

**Comment by tonybach** [comment]
Should probably a handle long error message when it overflows, not sure how exactly we should go about it though.

screen shot 2017-03-28 at 2 45 55 pm

File: cdap-ui/app/cdap/components/DataPrep/Directives/Filter/index.js (hunk 3 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/DataPrep/Directives/Filter/index.js:174]: Added the autofocus

**Comment by tonybach** [comment]
Code LGTM 👍

**Pull request merged**